engine/reformulation/core/src/main/java/it/unibz/inf/ontop/answering/reformulation/input/impl/RDF4JConstructQueryImpl.java

```diff
@@ -73,6 +73,8 @@ private static TupleExpr getFirstNonSliceExpression(TupleExpr expr) {
     }
 
     private static UnaryTupleOperator getFirstProjection(TupleExpr expr) {
+        if (expr instanceof QueryRoot)
```

Comment:
QueryRoot exists in RDF4J 3.x but isn't used. In RDF4J 4 parsed queries always start from QueryRoot so we need to unwrap them (needed in multiple places).